Neos.ContentRepository/Classes/Service/Utility/TransientNodeCache.php

```diff
@@ -0,0 +1,63 @@
+<?php
+namespace Neos\ContentRepository\Service\Utility;
+
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Simple cache for nodes that is intended to be used in `NodePrivilegeContext`.
+ *
+ * WARNING: This cache may contain nodes that should not be visible to the current user.
```

Comment:
If this class should only be used in NodePrivilegeContext I would suggest to move it somewhere int that folder structure.